akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowStatefulMapSpec.scala

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.scaladsl
+
+import akka.Done
+import akka.stream.testkit.StreamSpec
+import akka.stream.testkit.scaladsl.{ TestSink, TestSource }
+import akka.stream.{ ActorAttributes, Supervision }
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ Await, Promise }
+import scala.util.Success
+import scala.util.control.NoStackTrace
+
+class FlowStatefulMapSpec extends StreamSpec {
+
+  val ex = new Exception("TEST") with NoStackTrace
+
+  "A StatefulMap" must {
+    "work in the happy case" in {
+      val sinkProb = Source(List(1, 2, 3, 4, 5))
+        .statefulMap(() => 0)((agg, elem) => {
+          (agg + elem, Option((agg, elem)))
+        }, _ => None)
+        .collect {
+          case Some((agg, elem)) => (agg, elem)
+        }
```

Comment:
after change the `onComplete` from `Option[T]` to `T`, the `collect` part is always needed, it's more explicit but more boilerplate